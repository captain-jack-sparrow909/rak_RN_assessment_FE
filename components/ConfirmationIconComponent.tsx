import { View, Image, StyleSheet } from 'react-native';


const ConfirmationIconComponent = () => {
  return (
    <View>
      <Image
        source={require('../assets/success.png')}
        style={styles.icon}
      />
    </View>
  );
};

const styles = StyleSheet.create({
    icon: {
        width: 220,
        height: 220,
      },
});

export default ConfirmationIconComponent;
