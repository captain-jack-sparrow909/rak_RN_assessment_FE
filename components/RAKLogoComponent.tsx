import { View, Image, StyleSheet } from 'react-native';


const RAKLogoComponent = () => {
  return (
    <View>
      <Image
        source={require('../assets/rak_logo.png')}
        style={styles.icon}
        resizeMode="contain"
      />
    </View>
  );
};

const styles = StyleSheet.create({
    icon: {
        width: 220,
        height: 220,
      },
});

export default RAKLogoComponent;
