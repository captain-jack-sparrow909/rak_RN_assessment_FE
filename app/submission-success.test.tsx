import React from 'react';
import { render } from '@testing-library/react-native';
import SubmissionSuccess from './submission-success';
import RAKLogoComponent from '@/components/RAKLogoComponent';
import ConfirmationIconComponent from '@/components/ConfirmationIconComponent';

describe('SubmissionSuccess', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<SubmissionSuccess />);
    expect(toJSON()).toMatchSnapshot();
  });


  it('should render the success message and additional message', () => {
    const { getByText } = render(<SubmissionSuccess />);
    expect(getByText('Successfully Submitted!')).toBeTruthy();
    expect(getByText('Our representative will get in touch with you shortly')).toBeTruthy();
  });
});