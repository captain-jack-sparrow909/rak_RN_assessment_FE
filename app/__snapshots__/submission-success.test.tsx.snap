// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmissionSuccess should match the snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "marginTop": -120,
      }
    }
  >
    <View>
      <Image
        resizeMode="contain"
        source={1}
        style={
          {
            "height": 220,
            "width": 220,
          }
        }
      />
    </View>
  </View>
  <Text
    style={
      {
        "color": "#4CAF50",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Successfully Submitted!
  </Text>
  <View>
    <Image
      source={1}
      style={
        {
          "height": 220,
          "width": 220,
        }
      }
      testID="confirmation-icon"
    />
  </View>
  <Text
    style={
      {
        "color": "#888",
        "fontSize": 16,
        "marginTop": 10,
        "textAlign": "center",
      }
    }
  >
    Our representative will get in touch with you shortly
  </Text>
</View>
`;
